### PR TITLE
[SEMI-MODULAR] Variable Viewport

### PR DESCRIPTION
**Pull request by nevimer**
## About The Pull Request

This is a port of the variable viewport from Nebula. I've adapted it a bit in order to work with this codebase. It's not 100% done yet, there are still some issues. This mostly for testing and review, thought I would love a testmerge. 

Also. Whhhyyyy did it branch off an old failed branch. Whhhhhhyyyy.

https://user-images.githubusercontent.com/77420409/145335029-826f7ecb-eb4e-4526-8c01-36c2423c2073.mp4

## How This Contributes To The Skyrat Roleplay Experience

This is a massive QOL change for those who want in-between sizes for their chat and other GUI elements. I've restricted it to be one extra tile on the X and Y from previous so in-between sizes at 16:9 are not janky. 

## Changelog

:cl:
add: Viewport scale buttons.
qol: Viewport now auto resizes.
code: Changes absolute HUD vars to relative
refactor: Makes resize procs use the now client.onResize()
server: new config file options for max view range
/:cl:

### CODE REVIEW

File: code/_onclick/hud/_defines.dm [hunk 1]
```diff
@@ -9,190 +9,191 @@
 	"1:2,3:4" is the square (1,3) with pixel offsets (+2, +4); slightly right and slightly above the turf grid.
 	Pixel offsets are used so you don't perfectly hide the turf under them, that would be crappy.
 
-	In addition, the keywords NORTH, SOUTH, EAST, WEST and CENTER can be used to represent their respective
+	In addition, the keywords NORTH, BOTTOM, RIGHT, LEFT and CENTER can be used to represent their respective
```
Comment (tf-4) [inline code/_onclick/hud/_defines.dm:12]: Why did you rename all of these
Also put Skyrat edit comments

File: code/datums/brain_damage/split_personality.dm [hunk 2]
```diff
@@ -1,6 +1,6 @@
 #define OWNER 0
 #define STRANGER 1
-
+//SKYRAT EDIT START - Fixes split personality
```
Comment (tf-4) [inline code/datums/brain_damage/split_personality.dm:3]: ???

File: modular_skyrat/modules/viewport/client.dm [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+		set category = "Debug"
+
+		var/divisor = text2num(winget(src, "mapwindow.map", "icon-size")) || world.icon_size
+		var/winsize_string = winget(src, "mapwindow.map", "size")
+
+		to_chat(usr, "Current client view: [view]")
+		to_chat(usr, "Icon size: [divisor]")
+		to_chat(usr, "xDim: [round(text2num(winsize_string) / divisor)]")
+		to_chat(usr, "yDim: [round(text2num(copytext(winsize_string,findtext(winsize_string,"x")+1,0)) / divisor)]")
+	// proc/set_skybox_offsets(var/x_dim, var/y_dim) // No skybox on TG code.
```
Comment (tf-4) [inline modular_skyrat/modules/viewport/client.dm:64]: Just delete commented out code

**Review by tf-4 — COMMENTED** [review]
This should be done upstream

File: code/datums/brain_damage/split_personality.dm (hunk 2 above)
Comment (nevimer) [reply code/datums/brain_damage/split_personality.dm:3]: wrong branch it inherited from

**Comment by nevimer** [comment]
wrong branch ugh I hate git
